dotcom-rendering/src/web/components/BlockquoteBlockComponent.tsx

```diff
@@ -34,52 +35,108 @@ const quotedBlockquoteStyles = (palette: Palette) => css`
 	color: ${palette.text.blockquote};
 `;
 
-export const BlockquoteBlockComponent = ({ html, palette, quoted }: Props) => {
-	const {
-		willUnwrap: isUnwrapped,
-		unwrappedHtml,
-		unwrappedElement,
-	} = unwrapHtml({
-		fixes: [
-			{ prefix: '<p>', suffix: '</p>', unwrappedElement: 'p' },
-			{
-				prefix: '<blockquote>',
-				suffix: '</blockquote>',
-				unwrappedElement: 'blockquote',
-			},
-			{
-				prefix: '<blockquote class="quoted">',
-				suffix: '</blockquote>',
-				unwrappedElement: 'div',
-			},
-		],
-		html,
-	});
+const parseHtml = (html: string): DocumentFragment => JSDOM.fragment(html);
+
+// The nodeType for ELEMENT_NODE has the value 1.
+function isElement(node: Node): node is Element {
+	return node.nodeType === 1;
+}
+
+const getAttrs = (node: Node): NamedNodeMap | undefined =>
+	isElement(node) ? node.attributes : undefined;
+
+/**
+ * Searches through the siblings of an element to determine if it's the first
+ * of the desired node type.
+ */
+const isFirstSiblingOfType = (name: string, node: Node): boolean => {
+	const prevSibling = node.previousSibling;
+	if (!prevSibling) return true;
+	if (prevSibling.nodeName === name) return false;
+	else return isFirstSiblingOfType(name, prevSibling);
+};
 
-	if (quoted) {
-		const htmlWithIcon = unwrappedHtml
-			.trim()
-			.replace(
-				'<p>',
-				`<p>${renderToString(
-					<QuoteIcon colour={palette.fill.blockquoteIcon} />,
-				)}`,
-			);
-		return (
-			<RewrappedComponent
-				isUnwrapped={isUnwrapped}
-				html={htmlWithIcon}
-				tagName="blockquote"
-				elCss={quotedBlockquoteStyles(palette)}
-			/>
+const textElement =
+	(isQuoted: boolean, palette: Palette) =>
+	(node: Node, key: number): ReactNode => {
+		const text = node.textContent ?? '';
+		const children = Array.from(node.childNodes).map(
+			textElement(isQuoted, palette),
 		);
-	}
-	return (
-		<RewrappedComponent
-			isUnwrapped={isUnwrapped}
-			html={unwrappedHtml}
-			elCss={simpleBlockquoteStyles}
-			tagName={unwrappedElement}
-		/>
-	);
+		switch (node.nodeName) {
+			case 'P': {
+				// We want to add the quote icon to the first "P" node of the blockquote element
+				if (
+					isQuoted &&
+					node.parentElement?.nodeName === 'BLOCKQUOTE' &&
+					isFirstSiblingOfType('P', node)
+				) {
+					return (
+						<p>
+							<QuoteIcon colour={palette.fill.blockquoteIcon} />
+							{children}
+						</p>
+					);
+				}
+				return jsx('p', { children });
```

Comment:
Curious why we use the `jsx` function here instead of doing `<p>${children}</p>`